runtime/src/accounts_db.rs

```diff
@@ -6626,6 +6627,8 @@ impl AccountsDb {
                                             .cloned()
                                             .collect::<Vec<_>>();
                                         if !storages.is_empty() {
+                                            // metrics indicate the collect above over-allocates. This is almost always size 1
+                                            storages.shrink_to_fit();
```

Comment:
Are you looking at some system memory-related metrics? I went looking for how Vec does its default sizing (and/or any small-vec optimizations). Looks like we'll get [a vec with memory for 4 elements for normal uses of Vec](https://doc.rust-lang.org/src/alloc/raw_vec.rs.html#113-114). I do not know what happens with `.collect()` though. Did your metrics indicate a benefit for this `.shrink_to_fit()`?

Reply:
I removed it. I don't have time to look at it and it isn't that big of a deal. My metrics were initially 1 level too low and they returned that every slot's vector was size 4 when there was always only 1 item in it. It just offended my sense of right and wrong. But, it is more trouble than it is worth.